lib/linguist/heuristics.rb

```diff
@@ -274,7 +274,7 @@ def call(data)
     end
 
     disambiguate ".pl" do |data|
-      if /^(use v6|(my )?class|module)/.match(data)
+      if /^(use v6|(my)? class|module)/.match(data) #class might be a variable name in Perl 5
```

Comment:
> First, Any.pm which I have added in the test suite indeed is recognized as Perl6, which is it not. Since it does not have module or v6, it must be because line 49 fires that regular expression.

I ran the last released version of Linguist (`v4.7.3`) against `Any.pm` and it detected it as Perl5:
```
$ linguist Any.pm 
Any.pm: 101 lines (66 sloc)
  type:      Text
  mime type: application/x-pagemaker
  language:  Perl
```
Maybe the repository where that file is hasn't been updated in a long time...
<br>

> Please note that ^ might not be working as intended, since anyway class or module or use v6 might not be at the very beginning of a line, but indented by whitespace.

Good point. We could probably add something like `\s*` before `(my )?`.
<br>

> If that is not pressent, it fires whenever it finds class anywhere in a line (which indeed it does).

The four possible matching lines with `^(use v6|(my )?class|module)` are `^use v6`, `^my class`, `^class` and `^module`. If `class` is not at the beginning of the line it won't ever match.
<br>

> Anyway, experience is the mother of science. Please check the file I provide (Any.pm) with the former and current regex. If it checks out as perl6, then it's wrong. If it's perl5, then it's correct! :-)

I tried with https://regex101.com/ and it's not matching the Perl6 regular expression. It is, however, matching the Perl5 one.